python/ray/data/dataset.py

```diff
@@ -3424,6 +3424,9 @@ def __repr__(self) -> str:
     def __str__(self) -> str:
         return repr(self)
 
+    def __len__(self) -> int:
+        return self.count()
```

Comment:
Actually, could you add a simple test to `test_dataset`?

Reply:
Sure.

Reply:
https://github.com/ray-project/ray/pull/25152/commits/41e2dfbe8a1cd7d35edd08b7a07adbf3880d9085